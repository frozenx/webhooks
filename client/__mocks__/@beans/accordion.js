

import React from 'react';

const Accordion = (props) => (<div {...props} />);

export default { Accordion };