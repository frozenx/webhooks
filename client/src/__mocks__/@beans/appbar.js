import React from 'react';
const AppBar = (props) => (<div {...props}/>);

export default AppBar;
