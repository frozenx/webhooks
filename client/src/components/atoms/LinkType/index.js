import React, { Component, Fragment } from 'react';
import Dropdown from '@beans/dropdown';
import {
    InputType,
} from '../../../common/styles'
import LabelComponent from '../LabelComponent'

class LinkType extends Component {
    render() {
        const {
            attributeToRulesMapping,
            attr: {
                name,                               
                key,
                toolTip
            },
            id,
            values,
            errors,
            isFormLocked,
            linkableOptions,            
        } = this.props;
        return (
                <InputType>
                    <LabelComponent className='linkName' id={id} dark={true} emphasized={true} name={name} 
                    required={attributeToRulesMapping.required}></LabelComponent>
                    <Dropdown
                        name={name}
                        defaultValue='select'
                        id={id}
                        value={'select'}>                
                            <option value="select">Select</option>
                            {linkableOptions && linkableOptions.map((option, index) => {
                                return (
                                    <Fragment key={index}>
                                        <option value={option.name}>
                                            {option.name}
                                        </option>
                                    </Fragment>
                                )
                            })}
                    </Dropdown>
                </InputType>
        );
    }
}

export default LinkType;