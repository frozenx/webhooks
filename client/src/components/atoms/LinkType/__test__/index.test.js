import React from 'react';
import LinkType from '../';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { DefaultThemeProvider } from '@beans/theme';

let linkableOptions;

const getProps = props =>
  ({
    rules: 'regex string',
    errors: {},
    isDisabled: false,
    id: 1,
    linkableOptions: [{id: 1, name:'contact1', sites:[1, 2]}],
    ...props
  });

const setup = (additionalProps = {}) =>
renderer.create(
  <DefaultThemeProvider>
    <LinkType
    linkableOptions
    attr={{ id: 1, rules: [{ messages: [{ message: 'some message' }] }] }}
    {...getProps(additionalProps)}
    />
  </DefaultThemeProvider>
  ).toJSON();


describe('atoms/LinkType', () => {
    test('should create a snapshot', () => {
        const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
  test('should render the Contacts attribute for linkable contacts', () => {
    const Wrapper = shallow(<LinkType
      linkableOptions
      attr={{ id: 1, name:'Contacts', rules: [{ messages: [{ message: 'some message' }] }] }}
      {...getProps()}
    />);
    expect(Wrapper.find('LabelComponent').prop('name')).toEqual('Contacts');
  });
  test('should render the Addresses attribute for linkable address', () => {
    const Wrapper = shallow(<LinkType
      linkableOptions
      attr={{ id: 1, name:'Addresses', rules: [{ messages: [{ message: 'some message' }] }] }}
      {...getProps()}
    />);
    expect(Wrapper.find('.linkName').prop('name')).toEqual('Addresses');
  });
});