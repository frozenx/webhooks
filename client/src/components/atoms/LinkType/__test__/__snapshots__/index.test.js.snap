// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/LinkType should create a snapshot 1`] = `
<div
  className="sc-gzVnrw SKCKc"
>
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-fjdhpX gpWRWs"
  />
  <div
    className="base-components__RootElement-sc-150tvch-1 cCmuza"
  >
    <select
      className="base-components__BaseInput-sc-150tvch-2 beans-dropdown__select styled__Select-sc-10jppqs-1 GLMMI"
      defaultValue="select"
      disabled={false}
      id={1}
      onChange={[Function]}
      role="listbox"
    >
      <option
        value="select"
      >
        Select
      </option>
      <option
        value="contact1"
      >
        contact1
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="styled__AlignedIcon-sc-10jppqs-0 hHyXUl styled__StyledSVG-sc-1nulmu6-1 flhEYZ"
      color="currentColor"
      disabled={false}
      height={12}
      preserveAspectRatio="xMinYMax meet"
      viewBox="0 -1 20 12"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18.652 1.312L10 10.276 1.356 1.312"
        fill="none"
        stroke="#666666"
      />
    </svg>
  </div>
</div>
`;
